core/src/main/scala/kafka/log/Log.scala

```diff
@@ -1204,13 +1202,12 @@ class Log(@volatile private var _dir: File,
         // Do the read on the segment with a base offset less than the target offset
         // but if that segment doesn't contain any messages with an offset greater than that
         // continue to read from successive segments until we get some messages or we reach the end of the log
-        var done = segmentEntryOpt.isEmpty
+        var done = segmentOpt.isEmpty
         var fetchDataInfo: FetchDataInfo = null
+        val segmentsIterator = segmentOpt.map(segment => segments.higherSegments(segment.baseOffset))
```

Comment:
Given we can get baseOffset from segment, can we get rid of `floorEntry` overall? or make `floorEntry` private?

Reply:
Done, made it private.